CONTRIBUTING.md

```diff
@@ -18,45 +18,163 @@ any additional information for contributing code to this project.
+
+#### Running in a containerized environment
+
+To run specific tests, perform the following:
+
+1. Drop down into a test container
+
+```
+$ ./bin/test_integration -d
+```
+
+1. Under the individual test add the following:
+
+```
+<name-of-test-function>.<identifier>=True
+
+## Example
+def my-integration-test-function()
+  ...
+
+my-integration-test-function.someidentifier=True
+```
+
+1. Add the identifier of choice to the following command:
+
+```
+root@123456:/opt/conjur-api-python3# nose2 -v -X --config integration_test.cfg -A '<identifier>' $@
+
+## Example
+root@123456:/opt/conjur-api-python3# nose2 -v -X --config integration_test.cfg -A 'someidentifier' $@
+```
+
+1. You should see that only that specific test is run. Every change made locally can be seen in the container so you
+do *not* need to rebuild before running these tests again.
+
+#### Running as a process
+
+We provide the option to run the integration tests as a process. This means integration tests can run in the same way
+a user would use our CLI; without Python installed on the machine, using the Conjur CLI exec.
+
+The integration tests are wrapped in an integration_test_runner Python module to run in a Python-free environment.
+That way, tests can be run cross-platform.
+
+##### Setup
+
+1. Pack the `integration_test_runner.py` using PyInstaller in the platform to run the executable.
+
+  To pack: `pyinstaller -F test/util/test_runners/integration_test_runner.py`. Note that you will need
+  to pack each runner in each platform that you want to run the tests.
+
+1. Pack the Conjur CLI using PyInstaller in the platform to run the executable
+
+  To pack: `pyinstaller -F ./pkg_bin/conjur`. Note that you will need to pack each runner in each platform
+  that you want to run the tests. Also note that _only_ for macOS, you should pack the CLI as a directory instead of a
+  single file (using -D instead of -F). This will allow the CLI tests to run quicker.
+
+1. Run the executable: `./integration_test_runner`, supplying the below required parameters via the command line.
+
+###### Required parameters
+
+`--invoke_cli_as_process`, required to run the CLI as a process.
+
+`--cli_to_test`, path to the packed CLI executable to test against.
+
+Parameters like --url, --account, --login, --password, will used to configure the CLI. These values are used
+before each test profile is run to configure the CLI and run the integration tests successfully.
+
+`--identifier`, the test method with this identifier will be run (`integration` by default).
+To run as a process, the identifier should be `test_with_process`.
+
+`--files_folder` path to test assets (policy files, etc). This folder is located under `/test/test_config` in the
+repo. Copy this executable into every OS you wish to run the CLI integration tests.
+
+
+###### Example
+```
+./integrations_tests_runner -i test_with_process -u https://conjur-server -a someaccount -l somelogin -p Myp@ssw0rd1!
```

Comment:
can we use full-name flags?

Reply:
also i would break this per param for readability:
```
./integrations_tests_runner \
  -i test_with_process \
  -u https://conjur-server \
  -a someaccount \
  -l somelogin \
  -p Myp@ssw0rd1! \
  -f /test \
  -c /dist/conjur \
  --invoke_cli_as_process
```

Reply:
great idea for readability